src/docker/shared.rs

```diff
@@ -532,7 +531,15 @@ pub(crate) fn image_name(config: &Config, target: &Target) -> Result<String> {
         "main"
     };
 
-    Ok(format!("{CROSS_IMAGE}/{target}:{version}"))
+    if uses_zig {
+        if let Some(image) = config.zig_image(target)? {
+            Ok(image)
+        } else {
+            Ok(format!("{CROSS_IMAGE}/zig:{version}"))
+        }
+    } else {
+        Ok(format!("{CROSS_IMAGE}/{target}:{version}"))
+    }
```

Comment:
Because `zig.enable` might not be active even if `zig` has an image. It makes sense if `build.zig.image` is set, but `target.(...).zig.enable` is deactivated. But there's probably a better way to do this, probably just getting the target name.